modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/parse/MBObjectParser.java

```diff
@@ -1039,24 +1041,25 @@ private Expression string(String context, Object obj, String fallback) {
     public Expression string(JSONArray json, int index) {
         Object obj = json.get(index);
         if (obj == null) {
-            return ff.literal("");
+            return ff.literal(null);
         } else if (obj instanceof String) {
             String str = (String) obj;
             return ff.literal(str);
         } else if (obj instanceof Number) {
             Number number = (Number) obj;
-            return ff.literal(number.toString());
+            return ff.literal(number);
         } else if (obj instanceof Boolean) {
             Boolean bool = (Boolean) obj;
-            return ff.literal(bool.toString());
+            return ff.literal(bool);
         } else if (obj instanceof JSONObject) {
             MBFunction function = new MBFunction(this, (JSONObject) obj);
             return function.function(String.class);
         } else if (obj instanceof JSONArray) {
-            if (((JSONArray) obj).get(0) instanceof String) {
+            if (((JSONArray) obj).get(0) instanceof String &&
+                MBExpression.canCreate(((JSONArray) obj).get(0).toString())) {
                 return MBExpression.transformExpression((JSONArray)obj);
             } else {
-                throw new MBFormatException(context + " string from JSONArray not supported");
+                return ff.literal(obj);
```

Comment:
An exception should occur if there is a JSON array that isn't an instance of an MBExpression - but this is good as a placeholder until the "literal" type expression is implemented.